programs/system/src/lib.rs

```diff
@@ -1,5 +1,71 @@
 #![no_std]
 
+use pinocchio::{
+    account_info::AccountInfo,
+    instruction::Signer,
+    pubkey::Pubkey,
+    sysvars::{rent::Rent, Sysvar},
+    ProgramResult,
+};
+
+use crate::instructions::{Assign, CreateAccount, Transfer};
+
 pub mod instructions;
 
 pinocchio_pubkey::declare_id!("11111111111111111111111111111111");
+
+/// Create an account with a minimal balance to be rent-exempt.
+///
+/// The account will be funded by the `payer` if its current lamports
+/// are insufficient for rent-exemption.
+#[inline(always)]
+pub fn create_account_with_minimal_balance(
+    account: &AccountInfo,
+    space: usize,
+    owner: &Pubkey,
+    payer: &AccountInfo,
+    signers: &[Signer],
+    rent_sysvar: Option<&AccountInfo>,
+) -> ProgramResult {
+    let lamports = if let Some(rent_sysvar) = rent_sysvar {
+        let rent = Rent::from_account_info(rent_sysvar)?;
+        rent.minimum_balance(space)
+    } else {
+        Rent::get()?.minimum_balance(space)
+    };
+
+    if account.lamports() == 0 {
```

Comment:
what do you think about using hints from `pinocchio::hint` module?

i expect account is not exist, in the majority of cases
```suggestion
    if likely(account.lamports() == 0) {
```

Reply:
nice

Reply:
the same for the lib (i didn't test it in the real app).
As a part of library it compiles in the expected instructions.
Just wanted  to make extra sure this logic won't be miscompiled in the large program.